### ISSUE TITLE
Problem On Regenerating Results

### ISSUE THREAD

**Issue by chao1224** [deepchem/deepchem#827]
Hi there,

We are trying to reproduce the Kaggle results in paper [Is Mulitask Deep Learning Practical or Pharma?](http://pubs.acs.org/doi/abs/10.1021/acs.jcim.7b00146), but failed to reproduce same STL and MTL results.

One thing we want to confirm here is whether the STL and MTL results in the paper are using the same hyperparameter as specified here: [STL](https://github.com/deepchem/deepchem/blob/master/examples/kaggle/KAGGLE_tf_singletask.py#L33-L42) and [MTL](https://github.com/deepchem/deepchem/blob/master/examples/kaggle/KAGGLE_tf_robust.py#L30-L47)

Any help would be appreciated.

**Comment by rbharath** [comment]
Yes, I believe the code is the same, but the hyperparameters may be slightly different in code. Try running code using hyperparameters reported in the paper.

**Comment by chao1224** [comment]
@rbharath Thanks, just want to check again. So about the hyperparameter reported in the paper, do you mean the one in Figure. 1?

**Comment by rbharath** [comment]
I believe exact hyperparameters are in the SI. If you post more details on the failures you're seeing here, I can help debug.

**Comment by chao1224** [comment]
@rbharath Thanks. Just checked the SI, and I'm using the exact same structure, except batch size.

And on the test set, only `HIVINT` gets higher pearson score using MTL compared to STL.

**Comment by rbharath** [comment]
Try making sure every hyperparam matches exactly. The models are sensitive to hyperparmeter choices.

It's also possible that system differences (different TF version, different numpy, etc.) could be causing issues. You might need to do a new hyperparameter search using `dc.hyper`.

**Comment by chao1224** [comment]
@rbharath Thanks for explain. 

Another detaild question is which task/tasks you used for hyperparameter searching? Since in the SI, you mentioned both STL and MTL use the same hyperparameter.

**Comment by rbharath** [comment]
I'd recommend starting by making sure you can match previously published numbers on Kaggle with MTL. Our numbers and previous work from Ma et. al get mean multitask performance in the same perf range (with entirely different code bases). Try tuning the MTL until validation numbers are in that range.

**Comment by chao1224** [comment]
@rbharath Thanks again. Just for now, we have this following results:

| model | training pearson | test pearson |
| --- | --- | --- |
| STL | 0.87 | 0.44 |
| MTL | 0.
